server.go

```diff
@@ -712,15 +712,66 @@ func (s *Server) newHTTP2Transport(c net.Conn, authInfo credentials.AuthInfo) tr
 	return st
 }
 
+func floorCPUCount() uint32 {
```

Comment:
this is code repetition from the profiling core PR. I don't know where something like this should live so that it can be reused by both `google.golang.org/grpc` and `google.golang.org/grpc/internal/profiling` - any suggestions?